Update to v21.05 to align with SPDK v21.07

tanabarr opened the pull request:
- Use daos packaged python3-pyelftools RPM in CentOS7 build
- Patch to align commits with what SPDK DPDK submodule points to
- Use meson and ninja to build
- Add dh-python to Dockerfile to fix ubuntu build

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>
Co-authored-by: Brian J. Murrell <brian.murrell@intel.com>

tanabarr committed:
Update to v21.05 to align with SPDK v21.07

- Patch to align commits with what SPDK DPDK submodule points to
- Use meson and ninja to build

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

brianjmurrell commented on dpdk.spec:
```diff
@@ -226,7 +202,7 @@ done
 %if ! %{with tools}
 rm -rf %{buildroot}%{sdkdir}/usertools
-rm -rf %{buildroot}%{_sbindir}/dpdk-devbind
+rm -rf %{buildroot}%{_bindir}/dpdk-devbind.py
```
Why the change to `_bindir`?  This is a super-user tool isn't it?  Needs root to run?  If so, it should be in `_sbindir`

brianjmurrell commented on Jenkinsfile:
```diff
@@ -40,4 +40,4 @@
 
-packageBuildingPipelineDAOS(['distros': ['centos7', 'centos8', 'leap15']])
\ No newline at end of file
+packageBuildingPipelineDAOS(['distros': ['centos7', 'centos8', 'leap15']])
```
With all of the debian/ files being added, why is it not being added to this list to build?

Currently, DAOS is building on Ubuntu.  We don't want this to regress.

brianjmurrell commented on dpdk.spec:
```diff
@@ -297,28 +269,33 @@ sed -i -e 's:-%{machine_tmpl}-:-%{machine}-:g' %{buildroot}/%{_sysconfdir}/profi
 %endif
 
 %changelog
+* Fri Jun 21 2021 Tom Nabarro <tom.nabarro@intel.com> - 0:21.02-1
```
It seems a bit of time travel here.  :-)  When you get back, let me know what the winning lottery numbers were for that day.  :-D

That killed your PR right at linting unfortunately.

brianjmurrell commented on dpdk.spec:
```diff
@@ -147,11 +147,12 @@ as L2 and L3 forwarding.
+%autosetup -n dpdk-%{?commit0:%{commit0}}%{!?commit0:%{ver}} -p1
 
 %build
+pip install pyelftools
```
We cannot use `pip` inside RPM builds.  Instead you should add a `BuildRequires: python3-pyelftools` to get that RPM installed and provide _pyelftools_.

brianjmurrell commented on dpdk.spec:
```diff
@@ -147,11 +147,12 @@ as L2 and L3 forwarding.
-                              -Ddisable_drivers=compress/isal    \
+                              -Ddisable_drivers=compress/isal
 # sadly this results in an error
 #                              -Denable_docs=true
```
Is this still true?  We should try enabling this again and see if they have fixed their issues with it.

tanabarr replied:
```
[1154/1377] /usr/bin/python3 ../buildtools/call-sphinx-build.py /usr/bin/sphinx-build 21.02.0 /builddir/build/BUILD/dpdk-21.02/doc/guides /builddir/build/BUILD/dpdk-21.02/x86_64-redhat-linux-gnu/doc/guides
FAILED: doc/guides/html 
/usr/bin/python3 ../buildtools/call-sphinx-build.py /usr/bin/sphinx-build 21.02.0 /builddir/build/BUILD/dpdk-21.02/doc/guides /builddir/build/BUILD/dpdk-21.02/x86_64-redhat-linux-gnu/doc/guides
Traceback (most recent call last):
  File "../buildtools/call-sphinx-build.py", line 22, in <module>
    if StrictVersion(ver) >= StrictVersion('1.7'):
  File "/usr/lib64/python3.6/distutils/version.py", line 40, in __init__
    self.parse(vstring)
  File "/usr/lib64/python3.6/distutils/version.py", line 137, in parse
    raise ValueError("invalid version number '%s'" % vstring)
ValueError: invalid version number 'these.'
```

brianjmurrell commented on dpdk.spec:
```diff
@@ -297,28 +271,33 @@ sed -i -e 's:-%{machine_tmpl}-:-%{machine}-:g' %{buildroot}/%{_sysconfdir}/profi
 %endif
 %{_sysconfdir}/profile.d/dpdk-sdk-*.*
 %{_libdir}/*.so
+%{pmddir}/*.so
```
Surely these are unversioned DSOs that are needed at runtime, much like daos has in /usr/lib64/daos_srv/

brianjmurrell committed:
Update to build 21.05 (#16)

Add changelog entry and dh-python to Dockerfile
Apply packaging updates

PR-repos-el7: pyelftools@PR-1
PR-repos-el8: pyelftools@PR-1

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

tanabarr commented:
Gatekeeper: please use the PR commit's title and description when merging with master, thanks in advance
